include/mbedtls/ssl.h

```diff
@@ -1383,12 +1387,28 @@ struct mbedtls_ssl_config
 #endif
 
 #if defined(MBEDTLS_KEY_EXCHANGE_SOME_PSK_ENABLED)
+
 #if defined(MBEDTLS_SSL_SRV_C)
     /** Callback to retrieve PSK key from identity                          */
     int (*MBEDTLS_PRIVATE(f_psk))(void *, mbedtls_ssl_context *, const unsigned char *, size_t);
     void *MBEDTLS_PRIVATE(p_psk);                    /*!< context for PSK callback           */
 #endif
-#endif
+
+#if defined(MBEDTLS_SSL_CLI_C) && defined(MBEDTLS_SSL_PROTO_TLS1_3)
+    /** Callbacks to retrieve PSK key or stored session.                    */
+    void *MBEDTLS_PRIVATE(p_psk_list);
```

Comment:
Given the documentation of `mbedtls_ssl_conf_psk()` and `mbedtls_ssl_set_session()` saying respectively "If you attempt to register more than one PSK, this function fails, though this may change in future versions, which may add support for multiple PSKs." and "Calling this function multiple times will only be useful once TLS 1.3 is supported.", I think the plan is rather to keep using `mbedtls_ssl_conf_psk()` and `mbedtls_ssl_set_session()` to define multiple PSKs and resumption sessions/tickets on client side. To set multiple PSKs the client application would call several times `mbedtls_ssl_conf_psk()` resulting in a list of PSKs being registered in the configuration. Then, the TLS 1.3 client code would loop over this configuration list.

Reply:
- How about the order of psk and session tickets ? 
- About memory size, if reuse them, I prefer add a API the define the size of PSK and session ticket list.